src/masternodes/mn_checks.cpp

```diff
@@ -3788,6 +3800,64 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor {
         return mnview.AddProposalVote(obj.propId, obj.masternodeId, vote);
     }
 
+    Res operator()(const CTransferBalanceMessage &obj) const {
+        auto res = Res::Ok();
+
+        // owner auth
+        if (obj.type != CTransferBalanceType::EvmOut)
+            for (const auto &kv : obj.from) {
+                res = HasAuth(kv.first);
+                if (!res)
+                    return res;
+            }
```

Comment:
does EvmOut mean "sending from defichain to EVM" or "from EVM to defichain"?
cause this check implies that EvmOut means "from defichain to EVM"

but at line 3834  it looks like "EvmIn" is considered as "defichain to EVM". 
Maybe rename it DFCtoEVM and EVMtoDFC to prevent such confusion?